command/connect/envoy/envoy.go

```diff
@@ -321,6 +331,27 @@ func (c *cmd) lookupProxyIDForSidecar() (string, error) {
 	return proxyCmd.LookupProxyIDForSidecar(c.client, c.sidecarFor)
 }
 
+func (c *cmd) lookupGRPCPort() (int, error) {
+	self, err := c.client.Agent().Self()
+	if err != nil {
+		return 0, err
+	}
+	cfg, ok := self["DebugConfig"] // TODO: use string?
+	if !ok {
+		return 0, fmt.Errorf("no debug config") // TODO: error msg: what is acceptable to show to client?
+	}
+	port, ok := cfg["GRPCPort"]
+	if !ok {
+		return 0, fmt.Errorf("no grpc port") // TODO: error msg
+	}
+	portN, ok := port.(float64)
+	if !ok {
+		return 0, fmt.Errorf("non-float64 port number") // TODO: error msg
```

Comment:
```suggestion
		return 0, fmt.Errorf("invalid grpc port in agent response")
```
This is weird and jank message but it Shouldn't Happen(tm).